src/providers/profile/profile.ts

```diff
@@ -1277,7 +1280,7 @@ export class ProfileProvider {
     this.persistenceProvider.removeAllWalletData(walletId);
     this.events.publish('Local/WalletListChange');
 
-    return this.persistenceProvider.storeProfile(this.profile);
+    return this.persistenceProvider.storeProfile(this.profile.getObj());
```

Comment:
this is VERY error-prune.  Passing `this.profile` will also work and will lead to disastrous consequences.